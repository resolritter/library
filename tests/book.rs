use insta::assert_snapshot;
use std::{thread, time};
use stdext::function_name;
use test_utils::{format::format_test_name, read_snapshot, spawn_test_program, SpawnedTest};

#[async_std::test]
async fn test_get() -> std::io::Result<()> {
    let test_name = format_test_name(function_name!());
    let SpawnedTest {
        server_addr,
        mut process,
        log_dir,
    } = spawn_test_program();

    // TODO don't rely on sleep, use IPC
    thread::sleep(time::Duration::from_secs(5));
    let book_route = format!("{}/book/Rapunzel", &server_addr);

    let mut get = surf::get(book_route).await.unwrap();
    assert!(get.status() == 200);
    assert_snapshot!(test_name, get.body_string().await.unwrap());

    assert_snapshot!(read_snapshot(&log_dir));
    process.kill().unwrap();
    Ok(())
}
