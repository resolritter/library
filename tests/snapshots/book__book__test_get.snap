---
source: tests/book.rs
expression: get.body_string().await.unwrap()
---
{"data":{"id":2,"title":"Rapunzel"}}
